alacritty_terminal/src/grid/mod.rs

```diff
@@ -73,6 +73,34 @@ pub trait GridCell {
 }
 
 /// Represents the terminal display contents
+///
+/// ```notrust
+/// +--------------------------------------------+ <--- self.scroll_limit
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |               SCROLLUP REGION              |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// +--------------------------------------------+ <--- self.lines
```

Comment:
This is also a bit misleading, since `self.lines` is not the distance away from `zero`, but rather the distance from `display_offset`. It's of course difficult to show that in a graphic like this, but maybe something like the following could work out:

```
┌────────────────────────┐ <── max_scroll_limit + display_offset
│                        │
│      UNINITIALIZED     │
│                        │
├────────────────────────┤ <── scroll_limit + display_offset | raw.len
│                        │
│          ABOVE         │
│                        │
├────────────────────────┤ v lines
│                        │ │
│     VISIBLE REGION     │ │
│                        │ │
├────────────────────────┤ ^ <── display_offset
│                        │
│          BELOW         │
│                        │
└────────────────────────┘ <── zero
                         ^
                         |
                        cols
```